lib/lightdata.h

```diff
@@ -12,17 +12,58 @@ class lightdata {
   unsigned char fine = 0;
   unsigned char tdc = 0;
 
-  int chip() const { return index / 32; };
-  int eoch() const { return index % 64; };
-  int cindex() const { return tdc + 4 * index; };
-  
   lightdata(unsigned char _device, unsigned char _index, unsigned char _coarse, unsigned char _fine, unsigned char _tdc) :
     device(_device),
     index(_index),
     coarse(_coarse),
     fine(_fine),
     tdc(_tdc) { };
-    
+
+  int chip() const { return index / 32; };
+  int eoch() const { return index % 64; };
+  int cindex() const { return tdc + 4 * index; };
+  float time() const;
+
+  /** calibration **/
+
+  static float fine_if[16][768];
+  static float fine_cut[16][768];
+  static float fine_off[16][768];
+  static bool load_fine_calibration(std::string filename);
+
 };
 
+float lightdata::fine_if[16][768] = {0.};
+float lightdata::fine_cut[16][768] = {0.};
+float lightdata::fine_off[16][768] = {0.};
+
+float
+lightdata::time() const
+{
+  auto ci = cindex();
+  float corr = (float)fine * fine_if[device][ci] + fine_off[device][ci];
+  if (fine > fine_cut[device][ci]) corr -= 1.;
```

Comment:
Depending on how you define the correction, you are missing a factor of one clock cycle

Reply:
To elaborate: you will fit data with something like
x*[3]/[1] + [2] - (x>=[0])*[3]
where [0] is CUT, [1] is IF, [2] is OFF and [3] is fixed and equals to coarse_to_ns or whatever unit you are using for the 2D  seesaw histogram. If you phrase it like
(x/[1] + [2] - (x>=[0]))*[3]
 you can use it as you are in the equation with no multiplication

Reply:
parameter [3] does not exist
`time()` returns the time in clock cycles.
we will then define a `timens()` function that converts to `ns`